lib/blocktree/blocktree.go

```diff
@@ -330,6 +330,19 @@ func (bt *BlockTree) GetAllBlocks() []Hash {
 	return bt.root.getAllDescendants(nil)
 }
 
+// GetAllDescendants returns all blocks that are descendants of the given block.
+func (bt *BlockTree) GetAllDescendants(hash common.Hash) ([]Hash, error) {
+	bt.RLock()
+	defer bt.RUnlock()
+
+	node := bt.getNode(hash)
+	if node == nil {
+		return []common.Hash{}, fmt.Errorf("%w: for block hash %s", ErrNodeNotFound, hash)
```

Comment:
nit `nil` seems more appropriate/saves some allocation

```suggestion
		return nil, fmt.Errorf("%w: for block hash %s", ErrNodeNotFound, hash)
```